### ISSUE TITLE
Error when saving with "Replace $..."

### ISSUE THREAD

**Issue by Bjoern-Ge** [maths/moodle-qtype_stack#1007]
When I save a question and use "Replace $..." I get an error.
I'm unsure what the source of the error is.
The system is running php 8.0.29, moodle 4.1.4+ (Build: 20230804) and stack 2023072101

The error is:

Exception - Cannot access offset of type string on string
[More information about this error](https://docs.moodle.org/401/en/error/moodle/generalexceptionmessage)

Debug info:
Error code: generalexceptionmessage 

Stack trace:
    line 1451 of /question/type/questiontypebase.php: TypeError thrown
    line 149 of /question/type/stack/questiontype.php: call to question_type->import_or_save_files()
    line 538 of /question/type/questiontypebase.php: call to qtype_stack->save_question_options()
    line 85 of /question/type/stack/questiontype.php: call to question_type->save_question()
    line 294 of /question/bank/editquestion/question.php: call to qtype_stack->save_question()

**Comment by sangwinc** [comment]
Very sorry about this.  I've reproduced the problem, and will push a fix.  Thanks for reporting the issue here.

**Comment by daniil-berg** [comment]
Sorry to bump this, but we are on [`4.4.6`](https://github.com/maths/moodle-qtype_stack/commits/v4.4.6) (2023102700) and this error still occurs. It seems [this commit](https://github.com/maths/moodle-qtype_stack/commit/9169a587b644ecf1150859bcac18fa94b2110cad) never made it into the `master` branch.

**Comment by aharjula** [comment]
Unfortunately, I made a mess (with some security implications) and there was a need to release a quick bug-fix in the form of 4.4.6, as we did not have the time to merge in all the other fixes to the patch and test them together, this and many other commits were left to the next proper release. So there might be some fixes that were made after 4.4.5 and which are actually coming out in 4.4.7 (around the end of the year) and not in this "unscheduled" 4.4.6 release, this being one of them.

No doubt this commit would have been safe to fit into that release, but at that point priorities were different.
